File: frontend-react/src/services/SelectionService.js
import axios from 'axios';

const SELECTIONS_URL = "http://localhost:8081/api/v1/selections";

class SelectionService {

    addEmployeeToSelection = (employeeId) => {
        return axios.post(`${SELECTIONS_URL}/add-employee/${employeeId}`);
    }

    removeEmployeeFromSelection = (employeeId) => {
        return axios.delete(`${SELECTIONS_URL}/remove-employee/${employeeId}`);
    }

    clearSelection = () => {
        return axios.delete(`${SELECTIONS_URL}/clear`);
    }

    getCurrentSelection = () => {
        return axios.get(`${SELECTIONS_URL}/current`);
    }

    submitFinalSelection = () => {
        return axios.post(`${SELECTIONS_URL}/submit`);
    }

    getSelectionById = (selectionId, sortField, sortDirection, pageSize, pageNumber) => {
        return axios.get(`${SELECTIONS_URL}/${selectionId}`, {
            params: {
                sortField,
                sortDirection,
                pageSize,
                pageNumber
            }
        });
    };

    getSelectionByIdSearched = (selectionId, searchField, parameter, sortField, sortDirection, pageSize, pageNumber) => {
        return axios.get(`${SELECTIONS_URL}/${selectionId}`, {
            params: {
                searchField,
                parameter,
                sortField,
                sortDirection,
                pageSize,
                pageNumber
            }
        });
    };
}

export default new SelectionService();

File: backend-java/src/main/java/com/empire/employeefinder/service/SelectionService.java
package com.empire.employeefinder.service;

import com.empire.employeefinder.dto.response.SelectionResponseDto;

public interface SelectionService {

    SelectionResponseDto submitFinalSelection(String login);

    SelectionResponseDto addEmployeeToSelection(String login, Long employeeId);

    SelectionResponseDto removeEmployeeFromSelection(String login, Long employeeId);

    void clearSelection(String login);

    SelectionResponseDto getCurrentSelection(String login);

    SelectionResponseDto getById(Long selectionId, String searchField,
                                 String parameter, String sortField, String sortDirection,
                                 int pageSize, int pageNumber);
}
